### PR TITLE
feat(codebuild): adds report group type property

### LINKED ISSUE #14279: (CodeBuild Report Group): Defining the report type in CDK
When you create a new Report Group you should be able to select the report type (e,g,. Coverage)

There is no way to change the _**report type**_ from _**Test**_ to **_Coverage_** in props
` new ReportGroup(this.scope, 'test-coverage', {reportGroupName: 'test-coverage'});`

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by daschaa**
fixes #14279.
Introduces optional property to the ReportGroup to define the type of the report group (either 'TEST' or 'CODE_COVERAGE'). It just passes down the property to the underlying CfnReportGroup.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [x] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codebuild/lib/report-group.ts [hunk 1]
```diff
@@ -93,6 +93,16 @@ export interface ReportGroupProps {
...
+
+  /**
+   * The type of report group. This can be one of the following values:
+   *
+   * - **TEST** - The report group contains test reports.
+   * - **CODE_COVERAGE** - The report group contains code coverage reports.
+   *
+   * @default - TEST
+   */
+  readonly type?: 'TEST' | 'CODE_COVERAGE'
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-codebuild/lib/report-group.ts:105]: Please introduce an enum type

**Review by rix0rrr — DISMISSED** [review]

File: packages/@aws-cdk/aws-codebuild/lib/report-group.ts (hunk 1 above)
Comment (daschaa) [reply packages/@aws-cdk/aws-codebuild/lib/report-group.ts]: Done! I was not sure about the documentation tho. Is this ok? I tried to reuse to text which is in the [CloudFormation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-reportgroup.html). 
Thanks you for your review! ❤️

**Review by rix0rrr — APPROVED** [review]

**Comment by daschaa** [comment]
@rix0rrr Does this need to have a change in the README file?

**Comment by daschaa** [comment]
@rix0rrr I have now added the information to the package README file. Could you please review it again? Thanks in advance!

File: packages/@aws-cdk/aws-codebuild/README.md [hunk 2]
```diff
@@ -493,7 +493,32 @@ const project = new codebuild.Project(this, 'Project', {
...
 
-If you do that, you need to grant the project's role permissions to write reports to that report group:
+For a code coverage report, you can specify a report group with the code coverage report group type.
+
+```ts
+declare const source: codebuild.Source;
+
+// create a new ReportGroup
+const reportGroup = new codebuild.ReportGroup(this, 'ReportGroup', {
+    type: ReportGroupType.CODE_COVERAGE
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-codebuild/README.md:503]:
```suggestion
    type: codebuild.ReportGroupType.CODE_COVERAGE
```

**Review by comcalvi — DISMISSED** [review]

**Comment by daschaa** [comment]
@comcalvi Thanks for the hint about the documentation! Is this PR ready to be merged now? 😊

**Comment by daschaa** [comment]
@rix0rrr @comcalvi Can this be merged? :)

File: packages/@aws-cdk/aws-codebuild/test/report-group.test.ts [hunk 3]
```diff
@@ -142,4 +143,16 @@ describe('Test Reports Groups', () => {
       "UpdateReplacePolicy": "Delete",
     });
   });
+
+  test('can be created with type=CODE_COVERAGE', () => {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-codebuild/test/report-group.test.ts:147]: Please add a test here verifying that `type` defaults to `TEST`.

**Review by comcalvi — CHANGES_REQUESTED** [review]
Almost there! A small change and this should be ready to merge.

File: packages/@aws-cdk/aws-codebuild/test/report-group.test.ts (hunk 3 above)
Comment (daschaa) [reply packages/@aws-cdk/aws-codebuild/test/report-group.test.ts:147]: Done!

**Comment by daschaa** [comment]
@comcalvi I commited your suggestion regarding the JSDoc and implemented a test for the default report group type 😊 Could you review it again please? Thanks in advance! ❤️

**Comment by daschaa** [comment]
@comcalvi I changed the target brach to "main" - I hope it is ok, because I saw in other PR's it was done automatically..

**Review by comcalvi — APPROVED** [review]

**Pull request merged**